Makefile

```diff
@@ -38,153 +34,173 @@ help:  ## Display this help
 	@echo "  DEBUG            -- debug flag, if any ($(DEBUG))"
 
-.PHONY: $(KUSTOMIZE)
-$(KUSTOMIZE):
-	cd $(TOOLS_DIR); ./install_kustomize.sh
+# Image URL to use all building/pushing image targets
+IMG ?= baremetal-operator:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+## --------------------------------------
+## Test Targets
+## --------------------------------------
 
+# Run tests
 .PHONY: test
-test: fmt generate lint vet unit ## Run common developer tests
-
-.PHONY: show_packages show_dirs
-show_packages:
-	@echo $(PACKAGES)
-show_dirs:
-	@echo $(CODE_DIRS)
-
-.PHONY: generate
-generate: bin/operator-sdk ## Run the operator-sdk code generator
-	./bin/operator-sdk generate $(VERBOSE) k8s
-	./bin/operator-sdk generate $(VERBOSE) crds --crd-version=v1
-	openapi-gen \
-		--input-dirs ./pkg/apis/metal3/v1alpha1 \
-		--output-package ./pkg/apis/metal3/v1alpha1 \
-		--output-base "" \
-		--output-file-base zz_generated.openapi \
-		--report-filename "-" \
-		--go-header-file /dev/null
-
-bin/operator-sdk: bin
-	make -C tools/operator-sdk install
-
-bin:
-	mkdir -p bin
-
-.PHONY: travis
-travis: unit-verbose lint
+test: generate fmt lint vet manifests unit ## Run common developer tests
 
 .PHONY: unit
 unit: ## Run unit tests
-	go test $(GO_TEST_FLAGS) $(PACKAGES)
+	go test ./... $(VERBOSE) -coverprofile $(COVER_PROFILE)
 
 .PHONY: unit-cover
 unit-cover: ## Run unit tests with code coverage
-	go test -coverprofile=$(COVER_PROFILE) $(GO_TEST_FLAGS) $(PACKAGES)
+	go test -coverprofile=$(COVER_PROFILE) $(GO_TEST_FLAGS) ./...
 	go tool cover -func=$(COVER_PROFILE)
 
-.PHONY: unit-cover-html
-unit-cover-html:
-	go test -coverprofile=cover.out $(GO_TEST_FLAGS) $(PACKAGES)
-	go tool cover -html=cover.out
-
 .PHONY: unit-verbose
 unit-verbose: ## Run unit tests with verbose output
 	VERBOSE=-v make unit
 
+## --------------------------------------
+## Linter Targets
+## --------------------------------------
+
 .PHONY: linters
 linters: sec lint generate-check fmt-check vet ## Run all linters
 
-.PHONY: vet
-vet: ## Run go vet
-	go vet $(PACKAGES)
-
-.PHONY: lint
-lint: golint-binary ## Run golint
-	find $(CODE_DIRS) -type f -name \*.go  |grep -v zz_ | xargs -L1 golint -set_exit_status
-
-.PHONY: generate-check
-generate-check:
-	./hack/generate.sh
-
-.PHONY: generate-check-local
-generate-check-local:
-	IS_CONTAINER=local ./hack/generate.sh
-
 .PHONY: sec
 sec: $(GOPATH)/bin/gosec ## Run gosec
-	gosec -severity medium --confidence medium -quiet $(PACKAGES)
+	gosec -severity medium --confidence medium -quiet ./...
 
 $(GOPATH)/bin/gosec:
 	go get -u github.com/securego/gosec/cmd/gosec
 
+fmt: ## Run go fmt against code
```

Comment:
We're missing a lot of '.PHONY' targets.

Reply:
Fixing